scripts/bolt/internal/slurm_merge.sh

```
@@ -0,0 +1,175 @@
+                done
+            done
+            # Record only workloads that contributed profiles, so the manifest
+            # reflects what was ACTUALLY profiled (not the full suite declaration).
+            [ "$wl_had_fdata" -eq 1 ] && wls="$wls${wls:+,}$wl"
+        done
+        shopt -u nullglob   # scope nullglob to the gather loop: merge_fdata_files
+                            # relies on no-match globs yielding the literal pattern.
+        echo "[INFO] workloads profiled: $wls"
+        n=$(ls "$gathered" 2>/dev/null | wc -l)
+        echo "[INFO] gathered $n per-(workload,host) .fdata file(s)"
+        [ "$n" -gt 0 ] || { echo "[ERROR] no .fdata gathered from '"$FDATA_ROOT"'" >&2; exit 1; }
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)